homeassistant/helpers/aiohttp_client.py

```diff
@@ -138,17 +172,16 @@ async def async_aiohttp_proxy_stream(
 
 @callback
 def _async_register_clientsession_shutdown(
-    hass: HomeAssistantType, clientsession: aiohttp.ClientSession
+    hass: HomeAssistantType, clientsession_close: Callable
 ) -> None:
     """Register ClientSession close on Home Assistant shutdown.
 
     This method must be run in the event loop.
     """
 
-    @callback
-    def _async_close_websession(event: Event) -> None:
+    async def _async_close_websession(event: Event) -> None:
         """Close websession."""
-        clientsession.detach()
+        await clientsession_close()
```

Comment:
Note that we close now instead of detaching. Do we know why we used to only detach?

Reply:
I am not sure, but I think:

When HA shuts down, Python is also done. Detach will just make sure the clientsession notifies each connection that it's done and Python shutdown handles the rest?

https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientSession.detach

Reply:
So do we want to keep using `detach` or is it ok do use `close`? If we'll keep using detach, I guess we just patch `close` and then don't change the shutdown here.

Reply:
Yeah no need to change shutdown 👍

Reply:
```suggestion
        clientsession.detach()
```